conan/tools/microsoft/toolchain.py

```diff
@@ -148,7 +117,7 @@ def format_macro(key, value):
 
         cppstd = "stdcpp%s" % self.cppstd if self.cppstd else ""
         runtime_library = self.runtime_library
-        toolset = self.toolset
+        toolset = self.toolset or ""
```

Comment:
toolset uses string on next steps and None would be stringfyied to "None", which is not acceptable.